[weather] NPE if reqeuest timed out

The HttpUtil.executeUrl() returns `null`, if the request fails (see [documentation](https://www.eclipse.org/smarthome/documentation/javadoc/index.html)). Will provide a fix.

```
2017-10-13 12:18:59.718 [ERROR] [org.openhab.io.net.http.HttpUtil    ] - Fatal transport error: java.net.SocketTimeoutException: Rea
 timed out
2017-10-13 12:18:59.720 [ERROR] [nal.provider.AbstractWeatherProvider] - OPENWEATHERMAP: null
2017-10-13 12:18:59.721 [ERROR] [eather.internal.scheduler.WeatherJob] - null
java.lang.NullPointerException: null
        at org.openhab.binding.weather.internal.provider.AbstractWeatherProvider.executeRequest(AbstractWeatherProvider.java:104) [1
:org.openhab.binding.weather:1.11.0.201710020704]
        at org.openhab.binding.weather.internal.provider.AbstractWeatherProvider.getWeather(AbstractWeatherProvider.java:50) [10:org
openhab.binding.weather:1.11.0.201710020704]
        at org.openhab.binding.weather.internal.scheduler.WeatherJob.execute(WeatherJob.java:49) [10:org.openhab.binding.weather:1.1
.0.201710020704]
        at org.quartz.core.JobRunShell.run(JobRunShell.java:202) [111:org.eclipse.smarthome.core.scheduler:0.9.0.201710101404]
        at org.quartz.simpl.SimpleThreadPool$WorkerThread.run(SimpleThreadPool.java:573) [111:org.eclipse.smarthome.core.scheduler:0
9.0.201710101404]
```

--- a/bundles/binding/org.openhab.binding.weather/src/main/java/org/openhab/binding/weather/internal/provider/AbstractWeatherProvider.java
+++ b/bundles/binding/org.openhab.binding.weather/src/main/java/org/openhab/binding/weather/internal/provider/AbstractWeatherProvider.java
@@ -93,7 +93,7 @@ public abstract class AbstractWeatherProvider implements WeatherProvider {
         try {
             logger.trace("{}[{}]: request : {}", getProviderName(), locationConfig.getLocationId(), url);
 
-            String response = HttpUtil.executeUrl("GET", url, 15000);
+            String response = StringUtils.trimToEmpty(HttpUtil.executeUrl("GET", url, 15000));
 
             if (logger.isTraceEnabled()) {
                 response = StringUtils.remove(response, "\n");

